scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -309,154 +309,230 @@ For this example we will focus on identifying DE genes in these `Tumor_Myoblast`
-This allows us to use single-cell resolution to define the labels and sum gene counts across groups of cells containing the same label, and avoids counting each cell as its own replicate. 
+Before we can compare the gene expression profiles of myoblasts in ARMS vs. ERMS samples, we will need to "pseudo-bulk" the gene counts. 
+Pseudo-bulking creates a new counts matrix that contains the sum of the counts from all cells with a given label (e.g., cell type) for each sample ([Tung _et al._, 2017](https://doi.org/10.1038/srep39921)). 
+If we were to keep each cell's counts separate, treating them as replicate samples, we would end up with inflated statistics. 
+By pseudo-bulking first, will now have one count for each gene for each sample and we can take advantage of well-established methods for differential expression with bulk RNA-seq.
 
 Pseudo-bulking is implemented prior to differential expression analysis on single-cell data for the following reasons: 
 
-- Produces larger counts, which allows us to use standard normalization and differential expression methods used by bulk RNA-sequencing. 
--  Collapses gene expression counts by sample, so that samples, rather than cells, represent replicates.
+- Produces larger and less sparse counts, which allows us to use standard normalization and differential expression methods used by bulk RNA-seq. 
+- Collapses gene expression counts by sample, so that samples, rather than cells, represent replicates.
 - Masks variance within a sample to emphasize variance across samples.
-
-We are starting with the simple comparison of looking at one cell type and comparing across two groups of samples: the ERMS and ARMS subdiagnoses. 
-Our first step will be to pseudo-bulk our dataset to group our cells by cell type and by sample.
-This will create a new `SingleCellExperiment` (SCE) object that contains the pseudo-bulked counts for all genes. 
-We can subset this SCE to just contain our cell type of interest (tumor myoblasts) for input to differential expression analysis. 
+This can be both good and bad! 
+Masking intra-sample variation means you might miss identifying genes that have high intra-sample heterogeneity.
 
 Before we apply pseudo-bulking to our dataset, let's look at a simple example of how pseudo-bulking works. 
 
-```{r}
+```{r create matrix}
 # create an example counts matrix
-counts_mtx <- matrix(1:12, 
-                     ncol = 4,
-                     dimnames = list(c("geneA", "geneB", "geneC"),
-                                     c("A-cell1", "A-cell2", "B-cell1", "B-cell2")))
+counts_mtx <- matrix(
+  1:12, 
+  ncol = 4,
+  dimnames = list(c("geneA", "geneB", "geneC"),
+                  c("A-cell1", "A-cell2", "B-cell1", "B-cell2"))
+)
 counts_mtx
 ```
 
+Next we will create a pseudo-bulked matrix, with only 2 columns, 1 for group A and 1 for group B.
+To do this we will use the `DelayedArray::colsum()` function that allows us to sum the counts in all columns.
```

Comment:
It's confusing, but this actually sums counts for each row, across columns. This may be a question of how my brain works. 

Maybe: 

```suggestion
To do this we will use the `DelayedArray::colsum()` function, which allows us to sum the counts for each row across groups of columns.
```